lib/subproviders/geth_api_double.js

```diff
@@ -19,21 +19,31 @@ function GethApiDouble(options) {
   this.initialized = false;
 
   this.initialization_error = null;
+  this.post_initialization_callbacks = [];
 
   this.state.initialize(this.options, function(err) {
     if (err) {
       self.initialization_error = err;
     }
     self.initialized = true;
+
+    var callbacks = self.post_initialization_callbacks;
+    self.post_initialization_callbacks = [];
+
+    setImmediate(function() {
+      callbacks.forEach(function(callback) {
+        callback();
+      });
+    });
```

Comment:
I think this would be better as `callbacks.forEach(setImmediate)`

Reply:
Ah yes, that would prevent exceptions in one callback from preventing others from being run.

I'm still wrapping the `setImmediate()` call in a function though, to avoid accidentally forwarding the iteration arguments.

Reply:
Ah, right - good catch.